mandos/src/account.rs

```diff
@@ -61,12 +72,43 @@ impl CheckStorage {
 	}
 }
 
+#[derive(Debug)]
+pub enum CheckEsdt {
+	Star,
+	Equal(BTreeMap<BytesKey, CheckValue<BigUintValue>>),
+}
+
+impl InterpretableFrom<CheckEsdtRaw> for CheckEsdt {
+	fn interpret_from(from: CheckEsdtRaw, context: &InterpreterContext) -> Self {
+		match from {
+			CheckEsdtRaw::Star => CheckEsdt::Star,
+			CheckEsdtRaw::Equal(m) => CheckEsdt::Equal(
+				m.into_iter()
+					.map(|(k, v)| {
+						(
+							BytesKey::interpret_from(k, context),
+							CheckValue::<BigUintValue>::interpret_from(v, context),
+						)
+					})
+					.collect(),
+			),
+		}
+	}
+}
+
+impl CheckEsdt {
+	pub fn is_star(&self) -> bool {
+		matches!(self, CheckEsdt::Star)
+	}
+}
```

Comment:
You should also implement the Checkable trait here, it takes away code from other, more crowded places.